### ISSUE TITLE
cycle_ratio_example.cpp is non-deterministic and sometime fails

### ISSUE THREAD

**Issue by jzmaddock** [boostorg/graph#184]
We're running cycle_ratio_example.cpp in the tests, but it uses a non-deterministic random number generator which very occasionally results in the assertions failing.  It's not clear if this is inevitable, or is showing up a deeper bug of some sort.

Observed failure is:

```
Vertices number: 1000
Edges number: 30000
Maximum cycle ratio is 9.32637
Minimum cycle ratio is -0.45058
Critical cycle:
(354,386) (386,354) 
cycle_ratio_example: libs/graph/example/cycle_ratio_example.cpp:83: int main(int, char**): Assertion `std::abs(cr.first / cr.second - min_cr) < epsilon * 2' failed.
Aborted (core dumped)
```

Actually looking at the result, I suspect we just need to up the error tolerance slightly - 2 epsilon might be a bit hopeful?

**Comment by Becheler** [comment]
Mhhh I can not reproduce in current BGL after running the examples many times. It may be a CI host artifact, those changed along the years.

On a side note I have some afterthoughts with examples serving as both tests and examples, e.g. this assert:

https://github.com/boostorg/graph/blob/c1f4786e118311d45702c58e270046cb144cdde4/example/cycle_ratio_example.cpp#L79

The current effort the redo documentation #491  and examples may as well "solve" this issue by deprecating the example file

Closing this as stale, will revisit if anyone produces a fresh failure on current BGL ❤️

**Comment by jeremy-murphy** [comment]
RNG in unit tests is rarely a good thing. I say "rarely" because I know there are exceptions! ;)

**Comment by Becheler** [comment]
> RNG in unit tests is rarely a good thing. I say "rarely" because I know there are exceptions! ;)

Agreed ! You can use them to test random distributions invariants, but oh well even so it becomes a variance reproducibility challenge 😆
